src/Apple/Provider.php

```diff
@@ -134,24 +137,23 @@ public static function verify($jwt)
         $publicKeys = JWK::parseKeySet($data);
         $kid = $token->headers()->get('kid');
 
-        if (isset($publicKeys[$kid])) {
-            $publicKey = openssl_pkey_get_details($publicKeys[$kid]->getKeyMaterial());
+        if (!isset($publicKeys[$kid])) {
+            throw new InvalidStateException('Invalid JWT Signature');
+        }
+
+        $publicKey = openssl_pkey_get_details($publicKeys[$kid]->getKeyMaterial());
+        try {
             $constraints = [
-                new SignedWith(new Sha256, AppleSignerInMemory::plainText($publicKey['key'])),
+                new SignedWith(new Sha256, InMemory::plainText($publicKey['key'])),
```

Comment:
Switching from `AppleSignerInMemory` to `InMemory`, which provides checks against empty keys, and has (in more recent versions) also annotations for `#[SensitiveParameter]` to hide secrets.